notifications/src/main/kotlin/com/amazon/opendistroforelasticsearch/commons/notifications/model/NotificationConfig.kt

```diff
@@ -28,51 +31,46 @@ import org.elasticsearch.common.xcontent.XContentBuilder
 import org.elasticsearch.common.xcontent.XContentParser
 import org.elasticsearch.common.xcontent.XContentParserUtils
 import java.io.IOException
+import java.util.ArrayList
 import java.util.EnumSet
+import kotlin.jvm.Throws
 
 /**
  * Data class representing Notification config.
  */
 data class NotificationConfig(
-    val name: String,
-    val description: String,
-    val configType: ConfigType,
-    val features: EnumSet<Feature>,
-    val isEnabled: Boolean = true,
-    val slack: Slack? = null,
-    val chime: Chime? = null,
-    val webhook: Webhook? = null,
-    val email: Email? = null,
-    val smtpAccount: SmtpAccount? = null,
-    val emailGroup: EmailGroup? = null
+        val name: String,
+        val description: String,
+        val configType: NotificationConfigType,
+        val features: EnumSet<Feature>,
+        val isEnabled: Boolean = true,
+        val channelDataList: List<ChannelData?>,
```

Comment:
I think we don't require list here, we are interested in 1 config type per record. single item might reduce the complexity.